Whisper.net/SpeechToTextClient/WhisperSpeechToTextClient.cs

```diff
@@ -0,0 +1,89 @@
+// Licensed under the MIT license: https://opensource.org/licenses/MIT
+
+using System.Runtime.CompilerServices;
+using System.Text;
+using Microsoft.Extensions.AI;
+
+#pragma warning disable MEAI001 // Type is for evaluation purposes only and is subject to change or removal in future updates. Suppress this diagnostic to proceed.
+
+namespace Whisper.net;
+
+public sealed class WhisperSpeechToTextClient(string modelFileName) : ISpeechToTextClient
```

Comment:
Can we maybe allow different overloads to create a `WhisperSpeechToTextClient` ?

I think the best would be to allow passing the Factory directly => this way the clients can configure the Factory if they want to have some special usage for it (e.g. to use or not the GPU, etc). Or at least to accept [WhisperFactoryOptions](https://github.com/sandrohanea/whisper.net/blob/main/Whisper.net/WhisperFactoryOptions.cs) and some overload of providing the Memory<byte> for the ggml model (to not force the user to keep it on the disk). 

What do you think?

Reply:
> I think the best would be to allow passing the Factory directly => this way the clients can configure the Factory if they want to have some special usage for it (e.g. to use or not the GPU, etc). Or at least to accept [WhisperFactoryOptions](https://github.com/sandrohanea/whisper.net/blob/main/Whisper.net/WhisperFactoryOptions.cs) and some overload of providing the Memory for the ggml model (to not force the user to keep it on the disk).

That does make sense, I can add a specific contructor expecting those parameters to be provided from the caller for fine-grained control, absolutely.